File: app/src/main/java/com/softartdev/notecrypt/ui/settings/security/SecurityView.java
package com.softartdev.notecrypt.ui.settings.security;

import com.softartdev.notecrypt.ui.base.MvpView;

interface SecurityView extends MvpView {
    void showEncryptEnable(boolean encryption);
    void showPasswordDialog();
    void showSetPasswordDialog();
    void showChangePasswordDialog();

    interface DialogDirector {
        String getTextString();
        void showIncorrectPasswordError();
        void showEmptyPasswordError();
        void showPasswordsNoMatchError();
        void hideError();
    }
}

File: app/src/main/java/com/softartdev/notecrypt/ui/settings/security/SecurityView.kt
package com.softartdev.notecrypt.ui.settings.security

import com.softartdev.notecrypt.ui.base.MvpView

internal interface SecurityView : MvpView {
    fun showEncryptEnable(encryption: Boolean)
    fun showPasswordDialog()
    fun showSetPasswordDialog()
    fun showChangePasswordDialog()

    interface DialogDirector {
        val textString: String?
        fun showIncorrectPasswordError()
        fun showEmptyPasswordError()
        fun showPasswordsNoMatchError()
        fun hideError()
    }
}
